library/arith.lisp

```diff
@@ -480,6 +493,136 @@ The fields are defined as follows:
       (lisp Double-Float (x y)
         (cl:coerce (cl:/ x y) 'cl:double-float)))))
 
+(coalton-toplevel
+  (define-class (Trigonometric :a)
+    (sin   (:a -> :a))
+    (cos   (:a -> :a))
+    (tan   (:a -> :a))
+    (sinh  (:a -> :a))
+    (cosh  (:a -> :a))
+    (tanh  (:a -> :a))
+    (asin  (:a -> :a))
+    (acos  (:a -> :a))
+    (atan  (:a -> :a))
+    (asinh (:a -> :a))
+    (acosh (:a -> :a))
+    (atanh (:a -> :a)))
+
+  (declare sincos (Trigonometric :a => :a -> (Tuple :a :a)))
+  (define (sincos x)
+    (tuple (sin x) (cos x)))
+
+  (define-class (Exponentiable :a)
+    (**      (:a -> :a -> :a))
+    (logBase (:a -> :a -> :a)))
+
+  (declare sqrt ((Dividable Integer :a) (Exponentiable :a) => :a -> :a))
+  (define (sqrt x)
+    "Returns the square root of the input"
+    (** x (general/ (the integer 1) (the integer 2))))
+
+  (define-class ((Num :a) (Dividable :a :a) (Trigonometric :a) (Exponentiable :a) => Float :a)
+    (ee :a)
+    (pi :a))
+
+  (declare exp ((Float :f) => :f -> :f))
+  (define (exp x)
+    "Raise ee to the power of X"
+    (** ee x))
+
+  (declare log ((Float :f) => :f -> :f))
+  (define (log x)
+    "The natural logarithm of a given value"
+    (logBase ee x))
+
+  (define-class ((Float :f) (Ord :f) => RealFloat :f)
+    "Real Floating-Point Numbers"
+    (rationalize (:f -> Fraction)))
+
+  (declare atan2 ((RealFloat :f) => :f -> :f -> :f))
+  (define (atan2 y x)
+    "computes the arc tangent of y/x,
+     using the signs to determine the return value's quadrant"
+    (match (tuple (<=> x 0) (<=> y 0))
+      ((tuple (GT) _)    (atan (/ y x)))
+      ((tuple (LT) (LT)) (- (atan (/ y x)) pi))
+      ((tuple (LT) _)    (+ (atan (/ y x)) pi))
+      ((tuple (EQ) (GT)) (/ pi (fromint 2)))
+      ((tuple (EQ) (LT)) (/ pi (fromint -2)))
+      ((tuple (EQ) (EQ)) 0))))
+
+(cl:eval-when (:compile-toplevel :load-toplevel :execute)
+  (cl:defun %define-unary-wrapper
+      (return-type function cl:&key
+       (package :cl)
```

Comment:
i would nix these arguments; just double up with two arguments like `'sin 'cl:sin`, no need to match it like this